### DISCUSSION TITLE
Can we de-implement reaper's tax yet?

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#11204] [Ideas and suggestions]
Lets be honest, it's just been sitting around not doing anything ever since **"Among the Ancients"** reverted the improved "respawn with penalty" system it was supposed to be a part of.

As respawning is implemented currently: your character gets reset on death, like usual pre-reaper's-tax barotrauma - even all the way back to the pre-steam beta. So the "penalty" that reaper's tax is supposed to impart doesn't really make any sense now because it's a multiplicative penalty on top of a penalty that already sets your added skills to 0 (default), instead of the penalty to your OLD CHARACTER'S SKILLS as originally implemented.

Decide if you're going to bring back improved-respawns at some point, otherwise just get rid of this forgotten afterthought left over from a now-disused gameplay concept.

**Comment by Regalis11** [comment]
I hadn't noticed this post, but saw a link in https://github.com/Regalis11/Barotrauma/discussions/12055#discussioncomment-5982440 and decided I'd comment.

I don't entirely understand what you're saying here. Characters don't get reset on death, their skills just drop towards the initial skills (but not all the way down). Reaper's tax is not a penalty for dying, it's a penalty for respawning mid-round.

> "Among the Ancients" reverted the improved "respawn with penalty" system it was supposed to be a part of.

What do you mean by this? Unless I'm missing something, it didn't revert anything: the were no changes to the "respawn with penalty" logic, the only change was that your skill levels always drop a bit when you die.

> a penalty that already sets your added skills to 0 (default), instead of the penalty to your OLD CHARACTER'S SKILLS as originally implemented.

It doesn't set your added skills to 0, it lerps them 75% of the way to the maximum initial skill level. I also don't understand what you mean by a penalty to the old character's skills.

> otherwise just get rid of this forgotten afterthought left over from a now-disused gameplay concept.

Which disused gameplay concept do you mean?

**Reply by pantherpusher** [reply]
I think making skill penalty for dying/respawn an optional setting would be a good idea, considering how unfun it is to have to grind your skills back up, if you were at 100 in everything and did a midround respawn you'd be losing a lot of time.

**Reply by Regalis11** [reply]
To be clear, you don't need to do any grinding to get your skills back up if you respawn mid-round and receive Reaper's Tax, all you need to do is heal it at an outpost. The permanent skill loss from dying is another matter though, and I think we could consider making it optional. Although, personally I think dying should come with a significant cost, and I wouldn't mind seeing an even harsher penalty. :) That might be coming if/when we implement permadeath though.

**Reply by pantherpusher** [reply]
I also think death needs to have considerable cost, but I mostly play singleplayer so to me, death is a game over. I don't consider my crew replaceable. Still I can definitely see permadeath having a place in a more public sorta campaign where the crew gets swapped out with new players until either victory or the whole submarine goes down at once, if you can't make it to the next outpost you can't hire new crew.

**Reply by Anarengon** [reply]
>Still I can definitely see permadeath having a place in a more public sorta campaign where the crew gets swapped out with new players until either victory or the whole submarine goes down at once,

If you join on a server and play and die your rewarded with being kicked out of the server? Sounds like hell on earth. Unless you're the serverhost.

**Reply by rekterakathom** [reply]
Perhaps the permanent skill loss can be made into an option at the start of the campaign, similar to starting cash and difficulty. A simple slider from 0% to 100% skill loss would be simple enough and allows everyone to tweak it to their liking. 

It's probably not a huge deal to implement, but has a big effect on the gameplay. We are running a campaign with a really small crew and we feel like the 75% skill loss is too big of a hit and just restart the round if anyone dies, which I think greatly deducts from our experience.

**Reply by Regalis11** [reply]
> If you join on a server and play and die your rewarded with being kicked out of the server? Sounds like hell on earth. Unless you're the serverhost.

I agree, I don't think that'd work well. By permadeath I just meant losing everything (skills, talents) when you die.

>Perhaps the permanent skill loss can be made into an option at the start of the campaign, similar to starting cash and difficulty. A simple slider from 0% to 100% skill loss would be simple enough and allows everyone to tweak it to their liking.

I agree with this too, and actually went ahead and implemented it ([PR in our private repo](https://github.com/Regalis11/Barotrauma-development/pull/4336)).

**Reply by Ek-Videogames** [reply]
Yeah a server setting for 1. new character on death, and 2. reaper's tax loss percentage would allow a host's-end fix to that.

- For clarification, I'm talking about the fact that skills are set to the xml-defined random number ranges _(aka "new character")_ on death - and procedurally thereafter an additional -30% _(or whatever the current value of reaper's tax is)_ is applied on. So if you choose to respawn during a round you don't spawn with 60% of your stats before you died, you spawn with 60% of **beginning character stats**.
Odd logical choices aside - wouldn't it be more elegant either way to just be rid of reaper's tax entirely in favor of a single 'post-death' stat modifier, be that either 'set to default' or '% of last stats at time of death'?

- Under the briefly-existing implementation of reaper's tax, your character's current stats would be the same after you respawn - subtracted by the _(30% etc)_ leaving you with 60% of **your character's stats**. _(and then having to pay a station to cure you.)_ There was also benefit to choosing not to respawn during the round as the game wouldn't give you the stat debuff from reaper's tax in that case.

**Reply by Regalis11** [reply]
>1. new character on death, and 2. reaper's tax loss percentage would allow a host's-end fix to that.
>For clarification, I'm talking about the fact that skills are set to the xml-defined random number ranges (aka "new character") on death - and procedurally thereafter an additional -30% (or whatever the current value of reaper's tax is) is applied on. So if you choose to respawn during a round you don't spawn with 60% of your stats before you died, you spawn with 60% of beginning character stats.

I think there's still a slight misunderstanding here on how the penalties work. Dying makes the skills drop by 75% towards the initial skills, it doesn't drop them back to the initial skills nor create a new character. Reaper's Tax is added on top of that if you choose to respawn mid-round. A host also would not be able to remove/reduce the skill drop this if we just allowed choosing whether the player gets a new character or Reaper's Tax: they need to be able to adjust/remove the skill drop itself.

> Under the briefly-existing implementation of reaper's tax, your character's current stats would be the same after you respawn - subtracted by the (30% etc) leaving you with 60% of your character's stats. (and then having to pay a station to cure you.) There was also benefit to choosing not to respawn during the round as the game wouldn't give you the stat debuff from reaper's tax in that case.

This has not been changed, Reaper's Tax works the exact same way it did back then. In addition there is just that additional, permanent skill loss that you always get.

**Comment by ilmarilahti** [comment]
This has been resolved in the Blood in the Water update earlier this year. Reaper's Tax is gone, and instead there is now an additional configurable skill loss that is applied on top of the usual skill loss on death, if the player chooses to respawn mid-round rather than waiting for the round end. There's also a new "death prompt" that displays the actual net skill loss in detail.
